sdk/core/src/retry_policy.rs

```diff
@@ -0,0 +1,123 @@
+use tokio::time::{sleep, Duration};
+
+#[async_trait::async_trait]
+pub trait RetryPolicy {
```

Comment:
nit: I think we should get in the habit of documenting things so we don't make the lack of documentation worse than it is currently.

Reply:
Yes, at least I should document the logic behind the traits because it's expected for clients to implement their own retry policies (at least C# SDK allows it).